### PR TITLE
4 home page with nav

### PR DESCRIPTION
**Pull request by bob-bins**

### CODE REVIEW

File: ci [hunk 1]
```diff
@@ -88,26 +92,46 @@ docker:machine() { local args=$@
...
+docker:machine:host:isDefined() {
+    string:notEmpty "$(docker:machine ls --filter name=$(docker:machine:host:name) --quiet)"
 }
 
-docker:machine:createHost() {
-    set -e
-    if ! docker:machine:isHostDefined; then
-        docker:machine create "$(docker:machine:name)" --driver virtualbox
+docker:machine:host:isRunning() {
+    string:notEmpty "$(docker:machine ls --filter name=$(docker:machine:host:name) --filter state=Running --quiet)"
```
Comment (tastulae) [inline ci:104]: DRY with

```
docker:machine:host:existsWith() { local filterArgs=$@
   string:notEmpty "$(docker:machine ls --quiet $filterArgs)"
}
```

File: README.md [hunk 2]
```diff
@@ -1,24 +1,62 @@
 react-express-typescript
 ========================
 
-To get started, run `./ci` and see the options in the usage.
+This is a starting point template for a web ui that uses react, express, typescript technology stack, and
```
Comment (tastulae) [inline README.md:4]: that uses the react, express, typescript technology stack

File: README.md [hunk 3]
```diff
@@ -1,24 +1,62 @@
...
-1. Dist folders are duplicated somehow by webpack
\ No newline at end of file
+1. Dist folders are duplicated somehow by webpack
+
+# Release Notes
+
+## Version 1.0.0
+
+- Added hello world webpage
+- Added a single command to bring up an Express webserver and periodically compile Typescript code with tsc and webpack
```
Comment (tastulae) [inline README.md:62]: One-Line Install & Run Dockerized Build and Dev Environment for Mac
Hello World React Webpage running in Express

File: ci [hunk 4]
```diff
@@ -139,11 +163,27 @@ docker:run() { local imageName=$1; local tag=$2; local args=${@:3}
     docker:cli run "${imageName}:${tag}" ${args}
 }
 
-dev:environment:start() {
-    local tag="$(docker:build:tag:forLocalDev)"
-    local imageName="react-express-typescript-dev-environment"
-    docker:buildAndTag ${imageName} ${tag} ./Dockerfile .
-    docker:run ${imageName} ${tag}
+bash:runInBackground() { local logFile=$1; local command=${@:2}; export clearLogs
```
Comment (tastulae) [inline ci:166]: can we break up the items into several files instead of such a long file

File: ci [hunk 5]
```diff
@@ -139,11 +163,27 @@ docker:run() { local imageName=$1; local tag=$2; local args=${@:3}
...
+        rm "${logFile}"
+    fi
+     $command >> "${logFile}" 2>&1 &
+}
+
+dev:environment:runEverything() {
+    dev:environment:stopEverything
+    mkdir -p logs
+    dev:environment:docker:run:mounted "$(dev:environment:docker:baseImage)" npm i
+    dev:environment:docker:run:mounted "$(dev:environment:docker:baseImage)" echo "docker container is ready and working..."
```
Comment (tastulae) [inline ci:178]: need not do this given npm i happens first

File: ci [hunk 6]
```diff
@@ -171,11 +211,59 @@ dev:environment:express:run() {
 }
 
 docker:machine:ip() {
-    docker:machine inspect --format='{{.Driver.IPAddress}}' "$(docker:machine:name)"
+    docker:machine inspect --format='{{.Driver.IPAddress}}' "$(docker:machine:host:name)"
+}
+
+retry:until:returnsNonEmptyString() { local numberOfTimes=$1; local secondsBetweenTries=$2; local command=${@:3}
+    #TODO: do not output twice
```
Comment (tastulae) [inline ci:218]: remove todo

**Review by tastulae — CHANGES_REQUESTED** [review]

File: ci (hunk 4 above)
Comment (bob-bins) [reply ci:166]: i like the suggestion, matty, but not right now

**Review by tastulae — APPROVED** [review]

**Pull request merged**